src/transformers.jl

```diff
@@ -44,14 +44,12 @@ Does not need to be extended unless a mutating [`Transform`](@ref) is not possib
 function apply end
 
 """
-    apply!(A::AbstractArray{T}, ::Transform; dims=:, kwargs...) where T <: Real
+    apply!(A::AbstractArray, ::Transform; dims=:, kwargs...)
 
 Applies the [`Transform`](@ref) to each element of `A`.
 Optionally specify the `dims` to apply the [`Transform`](@ref) along certain dimensions.
 """
-function apply!(
-    A::AbstractArray{T}, t::Transform; dims=:, kwargs...
-) where T <: Real
+function apply!(A::AbstractArray, t::Transform; dims=:, kwargs...)
```

Comment:
I don't think this change is necessary given we can't mutate arrays for HoD, but maybe it's right in general. I don't know the reason it was restricted to `Real`.

Reply:
@glennmoy ?

Reply:
I was following Impute.jl's example.

Removing the type restriction seems ok to me.

We might still like to apply mutating transforms to non-numerical arrays, e.g. string, ZonedDateTimes, even though the ones we have currently implemented can't do that.